### PR TITLE
save project ID preference in Maven projects too

### LINKED ISSUE #731: App Engine project ID is not saved to preferences for new Maven based projects
#646 fixed this for native projects, but not for Maven-based project generation.

### PR DESCRIPTION
**Pull request by elharo**
@nbashirbello @akerekes fix #731 

This also moves some now common code into the preferences package. Going forward, we may be able to move some more code here, including constants for various preference IDs.

### CODE REVIEW

File: plugins/com.google.cloud.tools.eclipse.preferences.test/src/com/google/cloud/tools/eclipse/preferences/PreferenceUtilTest.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import org.junit.Test;
+
+public class PreferenceUtilTest {
+
+  private IProject project;
+  
+  @Before
+  public void setUp() {
+    IWorkspace workspace = ResourcesPlugin.getWorkspace();
+    project = workspace.getRoot().getProject("testproject" + Math.random());
```
Comment (akerekes) [inline plugins/com.google.cloud.tools.eclipse.preferences.test/src/com/google/cloud/tools/eclipse/preferences/PreferenceUtilTest.java]: Will the project be deleted after test execution?

File: plugins/com.google.cloud.tools.eclipse.preferences/src/com/google/cloud/tools/eclipse/preferences/PreferenceUtil.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.resources.ProjectScope;
+import org.eclipse.core.runtime.preferences.IEclipsePreferences;
+
+public class PreferenceUtil {
+
+  public static void setProjectIdPreference(IProject project, String projectId) {
+    if (projectId != null && !projectId.isEmpty()) {
+      IEclipsePreferences preferences = new ProjectScope(project)
+          .getNode("com.google.cloud.tools.eclipse.appengine.deploy");
```
Comment (akerekes) [inline plugins/com.google.cloud.tools.eclipse.preferences/src/com/google/cloud/tools/eclipse/preferences/PreferenceUtil.java]: We could extract this node ID into a constant.

Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.preferences/src/com/google/cloud/tools/eclipse/preferences/PreferenceUtil.java]: done

File: plugins/com.google.cloud.tools.eclipse.preferences.test/src/com/google/cloud/tools/eclipse/preferences/PreferenceUtilTest.java (hunk 1 above)
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.preferences.test/src/com/google/cloud/tools/eclipse/preferences/PreferenceUtilTest.java]: done

**Comment by elharo** [comment]
ping @akerekes

File: plugins/com.google.cloud.tools.eclipse.preferences/src/com/google/cloud/tools/eclipse/preferences/PreferenceUtil.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.resources.ProjectScope;
+import org.eclipse.core.runtime.preferences.IEclipsePreferences;
+
+public class PreferenceUtil {
+
+  private static final String DEPLOY = "com.google.cloud.tools.eclipse.appengine.deploy";
+
+  public static void setProjectIdPreference(IProject project, String projectId) {
+    if (projectId != null && !projectId.isEmpty()) {
```
Comment (briandealwis) [inline plugins/com.google.cloud.tools.eclipse.preferences/src/com/google/cloud/tools/eclipse/preferences/PreferenceUtil.java]: If null or empty, shouldn't we delete the preference key?

**Comment by elharo** [comment]
Ping @akerekes

File: plugins/com.google.cloud.tools.eclipse.preferences.test/src/com/google/cloud/tools/eclipse/preferences/PreferenceUtilTest.java (hunk 1 above)
Comment (akerekes) [inline plugins/com.google.cloud.tools.eclipse.preferences.test/src/com/google/cloud/tools/eclipse/preferences/PreferenceUtilTest.java]: We could use here `com.google.cloud.tools.eclipse.appengine.libraries.TestProject` if #771 were fixed.

**Pull request merged**